File: src/lib/types/word.ts
export interface Word {
  id: string;
  term: string;
  language: string;
  translation: string;
  phonetic?: string | null;
  explanation?: string | null;
  note?: string | null;
  tags: string[];
  source: string;
  createdAt: number;
  updatedAt: number;
}

export interface CreateWordRequest {
  term: string;
  language: string;
  translation: string;
  phonetic?: string | null;
  explanation?: string | null;
  note?: string | null;
  tags?: string[];
  source: string;
}

export interface UpdateWordRequest {
  id: string;
  term?: string;
  language?: string;
  translation?: string;
  phonetic?: string | null;
  explanation?: string | null;
  note?: string | null;
  tags?: string[];
  source?: string;
}

export interface ListWordsRequest {
  query?: string;
  tag?: string;
  limit?: number;
  offset?: number;
}

File: src-tauri/src/models/word.rs
// 单词相关请求类型

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateWordRequest {
    pub term: String,
    pub language: String,
    pub translation: String,
    pub phonetic: Option<String>,
    pub explanation: Option<String>,
    pub note: Option<String>,
    pub tags: Option<Vec<String>>,
    pub source: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateWordRequest {
    pub id: String,
    pub term: Option<String>,
    pub language: Option<String>,
    pub translation: Option<String>,
    pub phonetic: Option<String>,
    pub explanation: Option<String>,
    pub note: Option<String>,
    pub tags: Option<Vec<String>>,
    pub source: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListWordsRequest {
    pub query: Option<String>,
    pub tag: Option<String>,
    pub limit: Option<i32>,
    pub offset: Option<i32>,
}

